controllers/gateway_policy_discoverability_reconciler.go

```diff
@@ -14,5 +28,103 @@ func NewGatewayPolicyDiscoverabilityReconciler(client *dynamic.DynamicClient) *G
 }
 
 func (r *GatewayPolicyDiscoverabilityReconciler) Subscription() *controller.Subscription {
-	return &controller.Subscription{}
+	return &controller.Subscription{
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &machinery.GatewayGroupKind},
+			{Kind: &kuadrantv1beta3.AuthPolicyGroupKind},
+			{Kind: &kuadrantv1beta3.RateLimitPolicyGroupKind},
+			{Kind: &kuadrantv1alpha1.TLSPolicyGroupKind},
+			{Kind: &kuadrantv1alpha1.DNSPolicyGroupKind},
+		},
+		ReconcileFunc: r.reconcile,
+	}
+}
+
+func (r *GatewayPolicyDiscoverabilityReconciler) reconcile(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, _ *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("GatewayPolicyDiscoverabilityReconciler").WithName("reconcile")
+
+	gateways := lo.FilterMap(topology.Targetables().Items(), func(item machinery.Targetable, index int) (*machinery.Gateway, bool) {
+		g, ok := item.(*machinery.Gateway)
+		return g, ok
+	})
+
+	policyKinds := []*schema.GroupKind{
+		&kuadrantv1beta3.AuthPolicyGroupKind,
+		&kuadrantv1beta3.RateLimitPolicyGroupKind,
+		&kuadrantv1alpha1.TLSPolicyGroupKind,
+		&kuadrantv1alpha1.DNSPolicyGroupKind,
+	}
+
+	for _, gw := range gateways {
+		conditions := gw.Status.DeepCopy().Conditions
+
+		// TODO: What happens for multiple policies of the same kind -
+		// TODO: Should it be conditions per listener?
+		for _, policyKind := range policyKinds {
+			// Policies targeting gateway and listeners
+			path := []machinery.Targetable{gw}
+			path = append(path, topology.Targetables().Children(gw)...)
```

Comment:
I believe this will not work on Gateways with 2+ listeners. It would yield a path `Gateway → Listener 1 → Listener 2 → … → Listener N`